src/handlers/http/modal/ingest/ingestor_rbac.rs

```diff
@@ -55,13 +56,22 @@ pub async fn post_user(
 }
 
 // Handler for DELETE /api/v1/user/delete/{username}
-pub async fn delete_user(username: web::Path<String>) -> Result<impl Responder, RBACError> {
+pub async fn delete_user(
+    req: HttpRequest,
+    username: web::Path<String>,
+) -> Result<impl Responder, RBACError> {
     let username = username.into_inner();
     let _ = UPDATE_LOCK.lock().await;
     // fail this request if the user does not exists
     if !Users.contains(&username) {
         return Err(RBACError::UserDoesNotExist);
     };
+
+    // Deleting Correlations, Dashboards, Filters created by user
+    delete_correlations_filters_dashboards_for_user(&req)
+        .await
+        .map_err(|err| RBACError::Anyhow(anyhow::Error::msg(err.to_string())))?;
```

Comment:
Until we do the query side of #1143  we have to access multiple interfaces to perform this. In this case we should directly call `CORRELATIONS.delete`, `DASHBOARDS.delete`, etc.